mantidproject/mantid
# Resolve some deprecation warnings

**Description of work.**


So it appears that hiding warnings with using warnings filters only applies to the file you are adding the code to, rather than a global approach

Given that I've taken the approach of making the code in the specific reported situations work for new and old versions of matplotlib

**To test:**
THIS NEEDS TESTING on a new matplotlib (e.g. windows, ubuntu, mac) and old Red Hat

1. Open workbench
1. run the rest of this without seeing any warnings in the messages box (you might see one about collections, ignore that one.)
1. Run ```Load(Filename='MAR11060.raw', OutputWorkspace='MAR11060', SpectrumMin=100, SpectrumMax=200)```
1. Plot a colourfill plot
1. Click the setting icon
1. change the color map
1. Run ```Load(Filename='MAR11060.raw', OutputWorkspace='MAR11060', SpectrumMin=100, SpectrumMax=200)``` again
1. Run ```Load(Filename='MAR11060.raw', OutputWorkspace='MAR11060', SpectrumMin=1, SpectrumMax=20)``` and see the color fill plot change and update it's axes, especially the colour bar.
1. Close the plot, and show the slice viewer
1. change the colour map.
1. if you've got through this without warnings - success



Fixes #27863 re #27862. 

*This does not require release notes* because the warnings came in with python 3


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
